### PR TITLE
Fix failures from multiple inclusions of “x-test”.

### LINKED ISSUE #81: Multiple, top-level “x-test” imports cause silent failures.
## Context

When a test HTML file contained more than one `<script type="module">`, only the first script’s `it` / `describe` / `test` registrations actually ran. Registrations from later sibling scripts were silently dropped — no error, no warning, just a suite that mysteriously reported fewer tests than it declared.

One concrete issue here is that bundlers like Vite will _eagerly_ add imports into transpiled HTML to performance-optimize the import waterfall. This is how I discovered this issue in the first place. Even if this is a premature optimization, it is a sensible pattern that you (or your transpiler / bundler) might want to leverage.

## Proposal

We currently seal the registration window after a single microtask (`Promise.resolve()`). This only works in the case where the user has exactly one test module in their HTML and that test module imports `x-test` (statically, or through networked static imports).

Instead, we should only seal the registration window after `DOMContentLoaded`. Even if we consider the multi-import or multi-test case an _anti-pattern_ — supporting it is the only viable solution because we cannot easily detect / throw the alternative. I.e., it is better to support this than silently drop tests in this edge case.

**Note: This does _not_ enable the use of top-level awaits (see #80 for more details).**

## Example

```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">

    <!-- It should be fine if x-test is eagerly loaded in the HEAD -->
    <script type="module" src="../x-test.js"></script>

    <!-- Users should be allowed to load their tests (which also import “x-test”) in the HEAD -->
    <script type="module" src="test-boot-a.js"></script>
  </head>
  <body>
    <h3>test-boot</h3>

    <!-- Users should be allowed to load their tests (which also import “x-test”) in the BODY -->
    <script type="module" src="test-boot-b.js"></script>

    <!-- Users should be allowed to load multiple tests wherever (HEAD or BODY) -->
    <script type="module" src="test-boot-c.js"></script>

    <!-- Users should be allowed to load their tests as inlined modules (which also import “x-test”) in the BODY -->
    <script type="module">
      import { it, describe, assert } from '../x-test.js';

      describe('inline <script type="module">', () => {
        it('registers alongside src-backed scripts', () => {
          assert(true);
        });
      });
    </script>
  </body>
</html>
```

### PR DESCRIPTION
**Pull request by theengineear**
This change set moves our registration gate from being a microtask via `Promise.resolve()` to an await on `DOMContentLoaded`. This allows all imports (which are deferred as modules) to register via the standard `test` / `describe` / `it` calls _before_ the window closes.

It fixes the following concrete cases:

- Multiple test scripts included in HTML document (HEAD/BODY/inline).
- Performance optimizations to pull “x-test” into HEAD import.

Closes #81.

### CODE REVIEW

File: test/test-boot.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <!--
+      This suite documents how x-test boots across every supported
+      <script type="module"> arrangement: head placement, body placement,
+      multiple sibling scripts contributing to the same suite, and an
+      inline module script.
```
Comment (theengineear) [inline test/test-boot.html:9]: This is the crux test that was included to prove it all works. This fails (badly / silently) before we move from the microtask to `DOMContentLoaded`. Again, we cannot detect (and fail) for the reverse case, so our only real option is to _support this_ — and I think these are defensible things to do in a test HTML document anyhow.

File: x-test-common.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+export class XTestCommon {
```
Comment (theengineear) [inline x-test-common.js:1]: This helper class will be leveraged more in future PRs. Importantly, it now enables injection of the `domContentLoadedPromise`.

File: x-test-suite.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-export class XTestSuite {
-  static timeout = Symbol('timeout');
```
Comment (theengineear) [inline x-test-suite.js:2]: This was a code-smell. The `timeout` symbol was only ever here because there wasn’t a _better place_ to put it architecturally. Now we have `x-test-common.js`.

File: x-test-suite.js [hunk 4]
```diff
@@ -32,8 +32,8 @@ export class XTestSuite {
       XTestSuite.bail(context, event.reason);
     });
 
-    // Await a single microtask before we signal that we're ready.
-    XTestSuite.waitFor(context, Promise.resolve());
+    // Keep registration open until _at least_ DOMContentLoaded.
+    XTestSuite.waitFor(context, context.domContentLoadedPromise);
```
Comment (theengineear) [inline x-test-suite.js:36]: This is the crux change. We simply keep the registration window open longer. Again, this **does not** support top-level awaits, but it greatly improves how the runner executes.
